definition/definition_ser.go

```diff
@@ -392,6 +398,75 @@ func createLink(tasks map[string]*Task, linkRep *LinkRep, id int, ef expression.
 	return link, nil
 }
 
+func getLoopCfg(settings map[string]interface{}, ef expression.Factory) (*loopCfg, error) {
+	if len(settings) > 0 {
+
+		loop := &loopCfg{}
+
+		dowhile, ok := settings["do-while"]
+		if ok && dowhile != nil {
+			dowhileObj, err := coerce.ToObject(dowhile)
+			if err != nil {
+				return nil, fmt.Errorf("do-while configuration error : %s", err.Error())
+			}
+			condition, exist := dowhileObj["condition"]
+			if exist && len(condition.(string)) > 0 {
+				conditionStr := condition.(string)
+				if conditionStr[0] == '=' {
+					conditionStr = conditionStr[1:]
+				}
+				conditionExpr, err := ef.NewExpr(conditionStr)
+				if err != nil {
+					return nil, fmt.Errorf("compile do-while condition error: %s", err.Error())
+				}
+				loop.doWhile.condition = conditionExpr
+			}
+		}
+
+		retryonError, ok := settings["retryonerror"]
```

Comment:
Can you change the setting to camel case.

Reply:
Done. Changed `repeatonerror` to `repeatOnError` and `do-while` to `doWhile`